salt/modules/virt.py

```diff
@@ -454,6 +468,11 @@ def _get_disks(dom):
             qemu_target = source.get('file', '')
             if not qemu_target:
                 qemu_target = source.get('dev', '')
+            if source.get('file').startswith('/dev/zvol/'):
```

Comment:
The problem with ``source.get('file')`` is that you may get ``None`` if no there is no ``file``. Either you move that block above the ``if not qemu_target:`` and use ``qemu_target`` value or you need to default to the empty string.